addons/binding/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/handler/ChromecastHandler.java

```diff
@@ -102,43 +109,61 @@ private void createChromecast(final String address, final int port) {
             chromecast = new ChromeCast(address, port);
             chromecast.registerListener(this);
         } catch (Exception e) {
-            e.printStackTrace();
+            logger.error("Failed to connect ", e);
         }
     }
 
     private void destroyChromecast() {
-        chromecast.unregisterListener(this);
         try {
+            connectFuture.cancel(true);
+            cancelRefresh();
+
+            chromecast.unregisterListener(this);
             chromecast.disconnect();
         } catch (final IOException ex) {
             logger.debug("Disconnect failed: {}", ex.getMessage());
+        } finally {
+            chromecast = null;
         }
-        chromecast = null;
     }
 
     private void scheduleConnect(final boolean immediate) {
         final long delay = immediate ? 0 : 10;
-        if (futureConnect != null) {
-            futureConnect.cancel(true);
-            futureConnect = null;
+        if (connectFuture != null) {
+            connectFuture.cancel(true);
+            connectFuture = null;
         }
-        futureConnect = scheduler.schedule(() -> {
+
+        connectFuture = scheduler.schedule(() -> {
             try {
                 chromecast.connect();
-                // assume device is online as we no longer get notified
                 updateStatus(ThingStatus.ONLINE);
+                scheduleRefresh();
             } catch (final Exception e) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
                 scheduleConnect(false);
             }
         }, delay, TimeUnit.SECONDS);
     }
 
-    @Override
-    public void dispose() {
-        if (chromecast != null) {
-            destroyChromecast();
-        }
+    private void scheduleRefresh() {
+        cancelRefresh();
+        refreshFuture = scheduler.scheduleWithFixedDelay(this::handleRefresh, 0, 1, TimeUnit.SECONDS);
+    }
+
+    private void scheduleStopApp() {
+        scheduler.schedule(() -> {
+            try {
+                Application app = chromecast.getRunningApp();
+                if (app.id.equals(MEDIA_PLAYER) && this.appSessionId != null
+                        && app.sessionId.equals(this.appSessionId)) {
```

Comment:
You checked `this.appSessionId` to be null but `app.sessionId.equals(this.appSessionId)` will only fail if `app.sessionId` = null. 

So if `app.sessionId` cannot be null then the null check is not needed, otherwise your should check `app.sessionId` to be non-null.